src/compute/src/compute_state.rs

```diff
@@ -162,6 +164,16 @@ pub struct ComputeState {
     /// Interval at which to perform server maintenance tasks. Set to a zero interval to
     /// perform maintenance with every `step_or_park` invocation.
     pub server_maintenance_interval: Duration,
+
+    /// The [`mz_ore::now::SYSTEM_TIME`] at which the replica was started.
+    ///
+    /// Used to compute `replica_expiration`.
+    pub init_system_time: EpochMillis,
```

Comment:
I think this isn't precise, but might still work. As each _worker_ runs the code to initialize this independently, we can expect some skew between the individual workers, but since we should be crashing at the meet of the expiration times, this might not matter.

I don't have a good alternative at hands. Using the envd startup time is not an option, and neither is using the last deployment time. We want to retain the option of creating new replicas that'll then have a longer expiration than the one that's about to expire.

Reply:
The silver lining is that this discrepancy in replica_expiration for different workers would still be correct, but might just be surprising for someone looking at the expiration of multiple workers.